src/Microsoft.VisualStudio.Threading.Analyzers/CommonInterest.cs

```diff
@@ -84,8 +86,16 @@ internal static IEnumerable<TypeMatchSpec> ReadTypesAndMembers(AnalyzerOptions a
         {
             foreach (string line in ReadAdditionalFiles(analyzerOptions, fileNamePattern, cancellationToken))
             {
-                Match match = NegatableTypeOrMemberReferenceRegex.Match(line);
-                if (!match.Success)
+                Match? match = null;
+                try
+                {
+                    match = NegatableTypeOrMemberReferenceRegex.Match(line);
+                }
+                catch (RegexMatchTimeoutException)
+                {
+                }
+
+                if (match == null || !match.Success)
                 {
                     throw new InvalidOperationException($"Parsing error on line: {line}");
```

Comment:
I feel like we should report a timeout distinctly so folks realize it *could* just be that the computer was slow.

Reply:
Yeah, makes sense.